src/app/reports/run-report/run-report.component.ts

```diff
@@ -0,0 +1,405 @@
+      } else {
+          formattedResponse[new_key] = value['id'];
+      }
+    }
+    return formattedResponse;
+  }
+
+  /**
+   * Core run report functionality.
+   */
+  runreport() {
+    this.displayedColumns = [];
+    this.columnTypes = [];
+    this.isCollapsed = true;
+    const StartDate: Date = this.ReportForm.value.startDateSelect;
+    const EndDate: Date = this.ReportForm.value.endDateSelect;
+    // TODO: Update once language and date settings are setup
+    const dateFormat = 'yyyy-MM-dd';
+    this.ReportForm.patchValue({
+      startDateSelect: this.datePipe.transform(StartDate, dateFormat),
+      endDateSelect: this.datePipe.transform(EndDate, dateFormat)
+    });
+    const userResponse = this.formatUserResponse(this.ReportForm.value);
+    this.run_reportService.getRunReportData(this.report, userResponse).subscribe((res: any) => {
+      if ( this.report.type !== 'Pentaho') {
+        this.run_reportData = res.data;
+        res.columnHeaders.forEach((header: any) => {
+          this.columnTypes.push(header.columnDisplayType);
+          this.displayedColumns.push(header.columnName);
+        });
+      }
+      switch (this.report.type) {
+        case 'SMS':
+        case 'Table':
+            this.dataSource = new MatTableDataSource(this.run_reportData);
+            setTimeout(() => this.dataSource.paginator = this.paginator);
+            this.dataSource.sort = this.sort;
+            this.decimalPipe = `1.${this.decimalChoice.value}-${this.decimalChoice.value}`;
+            this.hideTable = false;
+          break;
+        case 'Chart':
+            this.run_reportData.forEach((object: any) => {
+              const temp: any = {};
+              temp.label = object.row[0];
+              temp.value = object.row[1];
+              this.inputData.push(temp);
+            });
+            this.setPieChart(this.inputData); // defaults to pie chart.
+            this.hideChart = false;
+          break;
+        case 'Pentaho':
+            const contentType = res.headers.get('Content-Type');
+            const file = new Blob([res.body], {type: contentType});
+            const filecontent = URL.createObjectURL(file);
+            this.pentaho_url = this.sanitizer.bypassSecurityTrustResourceUrl(filecontent);
+            this.hidePentaho = false;
+          break;
+      }
+    });
+  }
+
+  /**
+   * Checks the weather Mat-Table column has decimal display type.
+   * @param {number} index Index of column.
+   */
+  isDecimal(index: number) {
+     return this.columnTypes[index] === 'DECIMAL';
+  }
+
+  /**
+   *  Export to CSV functionality for report type "Table".
+   */
+  exportToCSV() {
+    const options = {
+      headers: this.displayedColumns
+    };
+    return new ngxCsv(this.run_reportData, `${this.report.name} Report`, options);
+  }
+
+  /**
+   * Prevents user from selecting end date before start date.
+   * @param {string} type startDate/endDate.
+   */
+  setMinDate(type: string) {
```

Comment:
Alternatives to this can be control error when start date > end date or end date < start date.